src/nupic/algorithms/Connections.cpp

```diff
@@ -418,78 +389,35 @@ bool segmentIncreasingOrder(const SegmentOverlap& a, const SegmentOverlap& b)
   return a.segment < b.segment;
 }
 
-void Connections::computeActivity(const vector<Cell>& input,
+void Connections::computeActivity(const vector<CellIdx>& input,
                                   Permanence activePermanenceThreshold,
                                   SynapseIdx activeSynapseThreshold,
                                   Permanence matchingPermanenceThreshold,
                                   SynapseIdx matchingSynapseThreshold,
-                                  vector<SegmentOverlap>& outActiveSegments,
-                                  vector<SegmentOverlap>& outMatchingSegments,
-                                  bool recordIteration)
+                                  vector<SegmentOverlap>& activeSegmentsOut,
+                                  vector<SegmentOverlap>& matchingSegmentsOut)
+  const
 {
-  vector<UInt32> numActiveSynapsesForSegment(nextFlatIdx_, 0);
-  vector<UInt32> numMatchingSynapsesForSegment(nextFlatIdx_, 0);
+  SegmentExcitationTally excitations(*this, activePermanenceThreshold,
```

Comment:
The link was to the SegmentExcitationTally docstring and to `ExtendedTemporalMemory::activateDendrites` where it computes the excitation using two different vectors.

This is also what I was referring to in the original comment:

> I aimed for clean code that avoids unnecessary computation. I avoided making copies of lists of cells -- I don't combine the "internal active cells" and "external active cells" into a separate "all active cells" list. This decision led me to rework the Connections computeActivity so that we could compute segment excitations without having to flatten all the active cells into a single list.

computeActivity takes in a vector, and is hence inflexible. The point of the SegmentExcitationTally is that it doesn't take in a data structure -- you do your own iteration with it.